plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseMySqlConnectorTest.java

```diff
@@ -337,6 +343,107 @@ public void testPredicatePushdown()
                 .isFullyPushedDown();
     }
 
+    @Test(dataProvider = "collation")
+    public void testPredicatePushdownWithCollationView(String charSet, String collation)
+    {
+        onRemoteDatabase().execute(format("CREATE OR REPLACE VIEW tpch.test_view AS SELECT regionkey, nationkey, CONVERT(name USING %s) COLLATE %s AS name FROM tpch.nation;", charSet, collation));
+        testNationCollationQueries("test_view");
+        onRemoteDatabase().execute("DROP VIEW tpch.test_view");
+    }
+
+    @Test(dataProvider = "collation")
+    public void testPredicatePushdownWithCollation(String charSet, String collation)
+    {
+        try (TestTable testTable = new TestTable(
+                onRemoteDatabase(),
+                "tpch.nation_collate",
+                format("AS SELECT regionkey, nationkey, CONVERT(name USING %s) COLLATE %s AS name FROM tpch.nation", charSet, collation))) {
+            testNationCollationQueries(testTable.getName());
+        }
+    }
+
+    private void testNationCollationQueries(String objectName)
+    {
+        // varchar inequality
+        assertThat(query(format("SELECT regionkey, nationkey, name FROM %s WHERE name != 'ROMANIA' AND name != 'ALGERIA'", objectName)))
+                .isFullyPushedDown();
+
+        // varchar like
+        assertThat(query(format("SELECT regionkey, nationkey, name FROM %s WHERE name LIKE 'ROMANIA'", objectName)))
```

Comment:
`LIKE ` with wild-card characters ? `LIKE 'ROMANIA` is equivalent to ` = 'ROMANIA`

Reply:
refactored

Reply:
You correct in case of LIKE with wild-card - it's transformed to >, < case and we don't support such cases

Reply:
Added support for like, please take a look